beluga/test/beluga/algorithm/test_estimation.cpp

```diff
@@ -252,5 +263,37 @@ TEST_F(ScalarEstimation, NonUniformWeightOverload) {
   ASSERT_NEAR(mean, 4.300, kTolerance);
   ASSERT_NEAR(standard_deviation, 2.026, kTolerance);
 }
+TEST_F(PoseCovarianceEstimation, MultiVariateNormalSE3) {
+  constexpr double kTolerance = 0.01;
+  const auto expected_mean =
+      Sophus::SE3d{Sophus::SO3d::exp(Eigen::Vector3d{-0.17, 0.25, 0.1}), Eigen::Vector3d{1.0, 2.0, 3.0}};
+  const Eigen::Matrix<double, 6, 6> expected_cov = Eigen::Matrix<double, 6, 6>::Identity() * 2e-1;
+  auto distribution = beluga::MultivariateNormalDistribution{expected_mean, expected_cov};
+
+  const auto samples =
+      beluga::views::sample(distribution) | ranges::views::take_exactly(500'000) | ranges::to<std::vector>;
+  const auto [mean, cov] =
+      beluga::estimate(samples, ranges::views::repeat_n(1.0, static_cast<std::ptrdiff_t>(samples.size())));
```

Comment:
Can we extend this to 3D?
https://github.com/Ekumen-OS/beluga/blob/cc6b52c5412612ebed74b9d10509179913d5e62d/beluga/include/beluga/algorithm/estimation.hpp#L281-L306